### DISCUSSION TITLE
Why do Weblate ask for pot/po files?

### DISCUSSION THREAD

**Discussion by buhtz** [WeblateOrg/weblate#8496] [Q&A] [answered]
I use a Weblate instance hosted by [Codeberg.org](https://codeberg.org).
I try to add a new translation component. This doesn't work because Weblate ask me for a po file.

In the section "Choose translation files to import" I have to options
 - Specify configuration manually
 - File format JSON nested structure file, File mask *.json

I don't know what that means in the context of translation. Do I have to add a weblate config file into my repo, first?

On the next site I'm asked for a "file format" and "file mask".

I don't understand what Weblate ask me here. The [repo I want to translate](https://codeberg.org/buhtz/spielwiese) does have a Python source file using importing `gettext` and using `_()` for strings.

Isn't weblate able to deal with that informations?

**Comment by nijel** [comment]
You need to create the PO/POT files, this usually involves invoking xgettext with some parameters, or intltool. See https://docs.weblate.org/en/latest/devel/gettext.html

**Reply by buhtz** [reply]
OK. Can we make an FeatureRequest for that? Weblate should do that by its own. I know it may not trivial and will take resources.

**Reply by nijel** [reply]
This is currently out of scope for Weblate - the extraction is typically more complex than just calling xgettext. In the end, this would need implementing a general CI service to execute arbitrary code the project needs to update the translations.

**Reply by buhtz** [reply]
What is more behind it then just xgettext?

**Reply by nijel** [reply]
* Gnome/Gtk apps typically use https://freedesktop.org/wiki/Software/intltool/
* For Qt you would use https://doc.qt.io/qt-6/linguist-manager.html#using-lupdate (this is not a PO file, but it serves the same purpose)
* Django wraps it with a Python code, see https://docs.djangoproject.com/en/4.1/ref/django-admin/#django-admin-makemessages
* Even when using xgettext natively, it is often wrapped with custom code to generate additional translatable files, for example in git: https://github.com/git/git/blob/57e2c6ebbe7108b35ba30184dcbcb6c34c929ad8/Makefile#L2884-L2930

These are the first ones I've remembered right now, but there are endless possibilities in this. That's why I don't see a simple solution to this… Such things are really better handled in generic CI environment, where you can trigger arbitrary actions based on changes in the files.

**Comment by buhtz** [comment, marked as answer]
Closing this because of follow-up question here:
https://github.com/WeblateOrg/weblate/discussions/8532
